ee/tabby-webserver/src/service/repository.rs

```diff
@@ -25,18 +37,57 @@ impl RepositoryService for DbConn {
     }
 
     async fn create_repository(&self, name: String, git_url: String) -> Result<ID> {
-        Ok(self.create_repository(name, git_url).await?.as_id())
+        Ok((self as &DbConn)
+            .create_repository(name, git_url)
+            .await?
+            .as_id())
     }
 
     async fn delete_repository(&self, id: &ID) -> Result<bool> {
-        Ok(self.delete_repository(id.as_rowid()?).await?)
+        Ok((self as &DbConn).delete_repository(id.as_rowid()?).await?)
     }
 
     async fn update_repository(&self, id: &ID, name: String, git_url: String) -> Result<bool> {
-        self.update_repository(id.as_rowid()?, name, git_url)
+        (self as &DbConn)
+            .update_repository(id.as_rowid()?, name, git_url)
             .await?;
         Ok(true)
     }
+
+    async fn search_files(
+        &self,
+        name: String,
+        path_glob: String,
+        top_n: usize,
+    ) -> Result<Vec<FileEntry>> {
+        let git_url = self.get_repository_git_url(name).await?;
```

Comment:
might use a filesystem based matching iterator and take top_n.

e.g https://crates.io/crates/ignore

Reply:
Since we can now match against the in-memory map, I don't think this would be an improvement anymore